Deploying_Seq2Seq_Model_with_Hybrid_Frontend.ipynb

```diff
@@ -0,0 +1,720 @@
+    "\n",
+    "\n",
+    "# Load model\n",
+    "# If loading on same machine the model was trained on\n",
+    "checkpoint = torch.load(loadFilename)\n",
+    "# If loading a model trained on GPU to CPU\n",
+    "#checkpoint = torch.load(loadFilename, map_location=torch.device('cpu'))\n",
+    "encoder_sd = checkpoint['en']\n",
+    "decoder_sd = checkpoint['de']\n",
+    "encoder_optimizer_sd = checkpoint['en_opt']\n",
+    "decoder_optimizer_sd = checkpoint['de_opt']\n",
+    "embedding_sd = checkpoint['embedding']\n",
+    "voc = Voc(corpus_name)\n",
+    "voc.__dict__ = checkpoint['voc_dict']\n",
+    "\n",
+    "\n",
+    "print('Building encoder and decoder ...')\n",
+    "# Initialize word embeddings\n",
+    "embedding = nn.Embedding(voc.num_words, hidden_size)\n",
+    "embedding.load_state_dict(embedding_sd)\n",
+    "# Initialize encoder & decoder models\n",
+    "encoder = EncoderRNN(hidden_size, embedding, encoder_n_layers, dropout)\n",
+    "decoder = LuongAttnDecoderRNN(attn_model, embedding, hidden_size, voc.num_words, decoder_n_layers, dropout)\n",
+    "# Load trained model params\n",
+    "encoder.load_state_dict(encoder_sd)\n",
+    "decoder.load_state_dict(decoder_sd)\n",
+    "# Use appropriate device\n",
+    "encoder = encoder.to(device)\n",
+    "decoder = decoder.to(device)\n",
+    "# Set dropout layers to eval mode\n",
+    "encoder.eval()\n",
+    "decoder.eval()\n",
+    "print('Models built and ready to go!')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert Model to Torch Script\n",
+    "\n",
+    "### Encoder\n",
+    "\n",
+    "As previously mentioned, to convert the encoder model to Torch Script, we use **tracing**. Tracing any module requires running an example input through the model's `forward` method and trace the computational graph that the data encounters. The encoder model takes an input sequence and a corresponding lengths tensor. Therefore, we create an example input sequence tensor `test_seq`, which is of appropriate size (MAX_LENGTH, 1), contains numbers in the appropriate range $[0, voc.num\\_words)$, and is of the appropriate type (int64). We also create a `test_seq_length` scalar which realistically contains the value corresponding to how many words are in the `test_seq`. The next step is to use the `torch.jit.trace` function to trace the model. Notice that the first argument we pass is the module that we want to trace, and the second is a tuple of arguments to the module's `forward` method.\n",
+    "\n",
+    "\n",
+    "### Decoder\n",
+    "\n",
+    "We perform the same process for tracing the decoder as we did for the encoder. Notice that we call forward on a set of random inputs to the traced_encoder to get the output that we need for the decoder. This is not required, as we could also simply create a tensor of the correct shape, type, and value range. We trace the model in the exact fashion that we traced the encoder.\n",
```

Comment:
- "This is not required, as we could also simply create a tensor of the correct shape, type, and value range" probably need to elaborate on why this is the case (we do not have any constraints on the values of the tensor because we don't have any ops that could fault on out-of-range inputs)